features/networking/multus-ipv6.feature

```diff
@@ -0,0 +1,49 @@
+Feature: Multus-CNI ipv6 related scenarios  
+  
+  # @author weliang@redhat.com
+  # @case_id OCP-28968
+  @admin
+  Scenario: IPv6 testing for OCP-21151: Create pods with multus-cni - macvlan bridge mode
+    # Make sure that the multus is enabled
+    Given the master version >= "4.1"
+    And the multus is enabled on the cluster
+    Given the default interface on nodes is stored in the :default_interface clipboard
+    And evaluation of `node.name` is stored in the :target_node clipboard
+    # Create the net-attach-def via cluster admin
+    Given I have a project
+    When I run oc create as admin over "<%= ENV['BUSHSLICER_HOME'] %>/testdata/networking/multus-cni/NetworkAttachmentDefinitions/IPv6/macvlan-bridge-v6.yaml" replacing paths:
+      | ["metadata"]["namespace"] | <%= project.name %>                                                                                                                                                                                                                             |    
+      | ["spec"]["config"]| '{ "cniVersion": "0.3.0", "type": "macvlan", "master": "<%= cb.default_interface %>","mode": "bridge", "ipam": { "type": "host-local", "subnet": "fd00::192:168:22:0/112", "rangeStart": "fd00::192:168:22:100", "rangeEnd": "fd00::192:168:22:200"} }' |
+    Then the step should succeed
+
+    # Create the first pod which consumes the macvlan custom resource
+    When I run oc create over "<%= ENV['BUSHSLICER_HOME'] %>/testdata/networking/multus-cni//Pods/IPv6/1interface-macvlan-bridge-v6.yaml" replacing paths:
+      | ["spec"]["nodeName"] | "<%= cb.target_node %>" |
+    Then the step should succeed
+    And a pod becomes ready with labels:
+      | name=macvlan-bridge-pod-v6 |
+    And evaluation of `pod.node_name` is stored in the :pod_node clipboard
+
+    # Check that the macvlan with mode bridge is added to the pod
+    When I execute on the pod:
+      | /usr/sbin/ip | -d | link |
+    Then the output should contain "net1"
+    And the output should contain "macvlan mode bridge"
+    When I execute on the pod:
+      | bash | -c | /usr/sbin/ip addr show net1 \| grep -Po 'fd00::192:168:22:[0-9a-fA-F]{1,4}' |
```

Comment:
GNU grep extended and PCRE regex have `[:xdigit:]` character class     `grep -o -E  [[:xdigit:]]{1,4}`  

https://www.gnu.org/software/grep/manual/html_node/Character-Classes-and-Bracket-Expressions.html

Reply:
@rbbratta  Can not make it work using the way you mentioned above:
[15:45:56] INFO> Exit Status: 2
      | bash | -c | /usr/sbin/ip addr show net1 \| grep -o -E [[:xdigit:]]{1,4} |
    Then the step should succeed                                                                                                                                                # features/step_definitions/common.rb:4
      the step failed (RuntimeError)
      /home/weliang/cucushift/Test1/verification-tests/features/step_definitions/common.rb:6:in `/^the step should( not)? (succeed|fail)$/'
      features/networking/multus-ipv6.feature:34:in `Then the step should succeed'

Update to use grep -Po 'fd00:dead:beef::[0-9a-fA-F]{1,4}', and it just works.

Reply:
ah, probably quotes `''` for `{}`, `grep -o -E '[[:xdigit:]]{1,4}'` otherwise bash expands the `{}` I guess.  always quote problems...

but it `[[:xdigit:]]` works with perl `-P` too.